contracts/index_executor/src/actions/ethereum/sygma.rs

```diff
@@ -0,0 +1,255 @@
+        deposit_data.extend(token_stats);
+        deposit_data.extend(recipient_len);
+        deposit_data.extend(recipient);
+
+        let fee_handler = Contract::from_json(
+            self.eth.clone(),
+            self.fee_handler_address,
+            include_bytes!("../../abi/SygmaBasicFeeHandler.json"),
+        )
+        .expect("Bad abi data");
+
+        let fee: (Uint, Address) = resolve_ready(fee_handler.query(
+            "calculateFee",
+            (
+                sender,
+                self.from_domain_id,
+                self.to_domain_id,
+                resource_id,
+                hex!("").to_vec(),
+                hex!("").to_vec(),
+            ),
+            None,
+            Options::default(),
+            None,
+        ))
+        .unwrap();
+
+        let mut fee_data = vec![0u8; 32];
+        fee.0.to_big_endian(&mut fee_data);
+        let nonzero_index = fee_data
+            .iter()
+            .position(|&x| x != 0)
+            .unwrap_or(fee_data.len() - 1);
+        let fee_data: Vec<u8> = fee_data[nonzero_index..].to_vec();
+
+        let bridge_params = (self.to_domain_id, resource_id, deposit_data, fee_data);
+
+        let bridge_func = self
+            .contract
+            .abi()
+            .function("deposit")
+            .map_err(|_| "NoFunctionFound")?;
+        let bridge_calldata = bridge_func
+            .encode_input(&bridge_params.into_tokens())
+            .map_err(|_| "EncodeParamError")?;
+
+        let token = Contract::from_json(
+            self.eth.clone(),
+            spend_asset,
+            include_bytes!("../../abi/erc20.json"),
+        )
+        .expect("Bad abi data");
+        let approve_params = (self.erc20_handler_address, spend_amount);
+        let approve_func = token
+            .abi()
+            .function("approve")
+            .map_err(|_| "NoFunctionFound")?;
+        let approve_calldata = approve_func
+            .encode_input(&approve_params.into_tokens())
+            .map_err(|_| "EncodeParamError")?;
+
+        Ok(vec![
+            Call {
+                params: CallParams::Evm(EvmCall {
+                    target: spend_asset,
+                    calldata: approve_calldata,
+                    value: U256::from(0),
+
+                    need_settle: false,
+                    update_offset: U256::from(36),
+                    update_len: U256::from(32),
+                    spend_asset,
+                    spend_amount,
+                    receive_asset,
+                }),
+                input_call: None,
+                call_index: None,
+            },
+            Call {
+                params: CallParams::Evm(EvmCall {
+                    target: self.contract.address(),
+                    calldata: bridge_calldata,
+                    value: U256::from(0),
+
+                    need_settle: true,
+                    update_offset: U256::from(36),
```

Comment:
If we want to change the spend amount of deposit, we need to change `token_stats`, where I think the offset of the `token_stats` in input data should be [4 bytes method-sig + 32 bytes domain-id + 32 bytes rid] = 68?

Reply:
Should be 164 according to experiment